### DISCUSSION TITLE
Issue HSS

### DISCUSSION THREAD

**Discussion by msdarici** [polarfire-soc/polarfire-soc-documentation#508] [HSS and U-Boot]
Firstly, I want to run AMP demos on the Icicle kit (main purpose). I prepared an image and I think I successfully downloaded it to the board. However, the boot manager gets stuck at "starting kernel." I saw this issue through MMUART0. After researching, I noticed this could be caused by the HSS version. I thought I could fix it by updating the HSS version, but I encountered many issues related to FPGENPROG. Initially, it couldn’t be detected, but I solved this by running SoftConsole as an administrator. Now, when I try to run it in Non-Secure Boot 1 mode, I get the error: "ERROR - Running the fpgenprog.exe PROGRAM action failed". My PC is offline, so I couldn't share a photo.

Thanks

**Comment by hughbreslin** [comment]
Hey @msdarici can you try using the FPExpress job included in the icicle kit reference design [releases](https://github.com/polarfire-soc/icicle-kit-reference-design/releases) (MPFS_ICICLE_BASE_DESIGN_2024_09.zip) - this also includes the HSS so you don't have to build it or anything. You can just launch FPExpress and program it.

Are you running on Windows or Linux btw to try and see why FPExpress is failing in softconsole?

**Comment by msdarici** [comment]
Thanks, problem was solved by the FPExress-way. Im working on Windows but I can't understand why softConsole is got the Error. Can I found any log files about run process.

**Reply by hughbreslin** [reply]
It could be a combination of things - could you check the softconsole.cmd (its in your softconsole install directory) - here you can set the path to FPGENPROG, there should be an example of the line to add in on line 17/18 of the file. If you uncomment that and make sure the path is correct you should be able to run FPGENPROG from a standard user account, I'm not sure why you need to run as admin but that could be causing some other issue so this should resolve your need to run as an administrator
